[FLINK-24718][avro] update AVRO dependency to latest version

RyanSkraba opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI to do that following [this guide](https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines#AzurePipelines-Tutorial:SettingupAzurePipelinesforaforkoftheFlinkrepository).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.


**(The sections below can be removed for hotfixes of typos)**
-->

## What is the purpose of the change

* Update AVRO dependency to the latest version

(Continuing on the work of https://github.com/apache/flink/pull/17623)

## Brief change log

  - Updated org.apache.avro:avro:1.10.0 to org.apache.avro:avro:1.11.0

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is already covered by existing tests, such as *flink-avro/*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: yes
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

RyanSkraba commented:
@flinkbot run azure

MartijnVisser reviewed (approved):
It looks good to me and I would like to get it merged, but I'm not sure if there's a preference to wait for the fix for https://issues.apache.org/jira/browse/AVRO-3273

@fapaul What do you think?

MartijnVisser commented:
@zentol I've discussed this briefly with @fapaul but we would like to get your opinion on it.

zentol commented:
I'm not really fond of the workaround. Does the avro-maven-plugin actually have to be in sync with the avro version?

MartijnVisser commented:
@RyanSkraba Can you elaborate on that?

MartijnVisser commented:
@RyanSkraba Since AVRO 1.11.1 is now released, could you update this PR?

RyanSkraba commented:
I'd love to see this in 1.16! I'm currently with limited connection, but I'll get this updated to 1.11.1 within the week.

RyanSkraba commented:
@flinkbot run azure

eshadei commented:
Hello Team,

Can you please confirm, when we are planning to release Flink 1.6 with Avro 1.11.1?

RyanSkraba commented:
Hello!  I've rebased, removed the maven workaround that we needed for 1.11.0 and squashed everything together.  There is a CI failure with TPC-DS e2e tests, but I can't seem to determine if it's related!

zentol commented:
See FLINK-28789 for the cause of the e2e failure.

zentol commented on flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/typeutils/AvroTypeExtractionTest.java:
```diff
@@ -109,9 +109,9 @@ void testSimpleAvroRead(boolean useMiniCluster, @InjectMiniCluster MiniCluster m
-                        + "\"type_date\": 2014-03-01, \"type_time_millis\": 12:12:12, \"type_time_micros\": 00:00:00.123456, "
-                        + "\"type_timestamp_millis\": 2014-03-01T12:12:12.321Z, "
-                        + "\"type_timestamp_micros\": 1970-01-01T00:00:00.123456Z, \"type_decimal_bytes\": \"\\u0007Ð\", "
+                        + "\"type_date\": \"2014-03-01\", \"type_time_millis\": \"12:12:12\", \"type_time_micros\": \"00:00:00.123456\", "
```
Can you elaborate on why these quotes were added? Were they rejected by Avro etc.

RyanSkraba replied:
Hello!  This was a fix introduced by AVRO-3124, cleaning up the `toString()` representation to look more like valid JSON.

My personal opinion is that nobody should parse a `toString()` return value...  but it has been useful in testing.

RyanSkraba commented:
Hello! For info, did this make it to the list for 1.16?

MartijnVisser commented:
@flinkbot run azure

MartijnVisser commented:
> Hello! For info, did this make it to the list for 1.16?

Not yet, but perhaps we can still squeeze it in :)
